pkg/gofr/handler.go

```diff
@@ -101,14 +91,21 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			err = gofrHTTP.ErrorRequestTimeout{}
 		}
 	case <-done:
-		if websocket.IsWebSocketUpgrade(r) {
-			// Do not respond with HTTP headers since this is a WebSocket request
-			return
-		}
+		handleWebSocketUpgrade(r)
 	case <-panicked:
 		err = gofrHTTP.ErrorPanicRecovery{}
 	}
 
+	// **Handle Custom Headers** if `result` is a `Response`.
+	if resp, ok := result.(response.Response); ok {
+		for key, value := range resp.Headers {
+			if w.Header().Get(key) == "" {
+				w.Header().Set(key, value)
+			}
+		}
+		result = resp.Data
+	}
```

Comment:
Maybe a comment explaining the reason behind this.

Here is my understanding:

There is a need to propagate the header to the caller.

Then explain existing headers are not overwritten.

For this exact reason, I would create a method in


```suggestion
	if resp, ok := result.(response.Response); ok {
		resp.PropagateHeaders(w)
		result = resp.Data
	}
```

With something like this in response.Response struct

```go
func(response.Response resp) PropagateHeaders(w *http.Response)
	for key, value := range resp.Headers {
		if w.Header().Get(key) != "" {
			// do not overwrite existing header
			continue
		}
		w.Header().Set(key, value)
	}
}
```

I wrote this on a phone, there might be some errors 😅

Reply:
@shashank-zopsmart @Umang01-hash what do you guys think, should we extract the setting of custom headers to another function or a simple if condition should suffice in the ServerHTTP func itself

Reply:
@rigved-telang extracting the setting of custom headers to another function is not a bad idea as it would lead to a more clear code and auto explain what the function is intended to do

Reply:
Hi @Umang01-hash apologies for the inactiivity on the issue, I came to my hometown for Diwali and thus was not able to work on it. I will make the required changes within 4-5 days as I will be vacationing for another day. Thanks.

Reply:
It's OK, but thanks for letting us know